src/Http/Controllers/Inertia/Concerns/ConfirmsTwoFactorAuthentication.php

```diff
@@ -77,7 +77,8 @@ protected function hasJustBegunConfirmingTwoFactorAuthentication(Request $reques
      */
     protected function neverFinishedConfirmingTwoFactorAuthentication(Request $request, $currentTime)
     {
-        return is_null($request->user()->two_factor_confirmed_at) &&
+        return ! array_key_exists('code', $request->session()->getOldInput()) &&
```

Comment:
@driesvints this change feels hacky but in my tests everything worked just fine.

Here I try to find out if the user just entered a 2FA confirmation code which got rejected.